libs/interpret-vision/src/lib/VisionExplanationDashboard/Controls/FlyoutObjectDetection.tsx

```diff
@@ -109,6 +204,9 @@ export class FlyoutObjectDetection extends React.Component<
     const classNames = flyoutStyles();
     const predictedY = getJoinedLabelString(item?.predictedY);
     const trueY = getJoinedLabelString(item?.trueY);
+
+    this.drawBoundingBoxes(item);
```

Comment:
drawBoundingBoxes is getting called on every render, which means you are re-initializing the editor every render.

Instead, move the editor initialization to a separate function and only call it when the component mounts.

In the "component did update" lifecycle method check if the image source has changed and reset the editor content source if it has.

Draw the bounding boxes once after the editor content source has loaded (i.e. the image has loaded in the editor) and also in the "component did update" method in case the bounding box prop data has changed.

Reply:
Agree with this! Just as a note, there isn't any reason (in the current implementation) that the image source should change. So it might not be necessary to update "component did update"